rbac_api_with_domains_test.go

```diff
@@ -267,3 +268,12 @@ func TestGetAllDomains(t *testing.T) {
 
 	testGetAllDomains(t, e, []string{"domain1", "domain2"})
 }
+
+func TestDomainHierarchical(t *testing.T) {
+	e, _ := NewEnforcer("examples/rbac_with_domains_hierarchical_model.conf", "examples/rbac_with_domains_hierarchical_policy.csv")
+	origin := e.GetNamedRoleManager("g").(*defaultrolemanager.RoleManager)
+	dest := e.GetNamedRoleManager("g2").(*defaultrolemanager.RoleManager)
+	origin.SetTenantManager(dest)
+
+	testDomainEnforce(t, e, "alice", "dom2", "data", "read", true)
```

Comment:
I think in terms of the nature of this test case, alice should be tested from domains 1, 2 and 3 since she has access to all of them.

Reply:
Added more tests